Add API reference section (#218)
mirumee/saleor-docs: website/docusaurus2-graphql-doc-generator/README.md

# GraphQL Documentation Generator for Docusaurus 2

[![npm](https://img.shields.io/npm/dt/@edno/docusaurus2-graphql-doc-generator?style=flat-square)](https://www.npmjs.com/package/@edno/docusaurus2-graphql-doc-generator)
[![Latest Version](https://img.shields.io/npm/v/@edno/docusaurus2-graphql-doc-generator?style=flat-square)](https://www.npmjs.com/package/@edno/docusaurus2-graphql-doc-generator)
[![GitHub license](https://img.shields.io/github/license/edno/docusaurus2-graphql-doc-generator?style=flat-square)](https://raw.githubusercontent.com/edno/docusaurus2-graphql-doc-generator/main/LICENSE)

This plugin generates a **Markdown documentation** from a **GraphQL schema**.

The documentation is generated for **Docusaurus 2** [docs feature](https://v2.docusaurus.io/docs/docs-introduction).

## Installation

### npm

```shell
npm install --save @edno/docusaurus2-graphql-doc-generator
```

### Yarn

```shell
yarn add @edno/docusaurus2-graphql-doc-generator
```

> Then you add it in your site's `docusaurus.config.js`'s plugins option:

Reviewer: Then you need to add it to the plugin options in `docusaurus.config.js` of your site.

> Then you add it in your site's `docusaurus.config.js`'s plugins option:

Reviewer: Then you need to add it to the plugin options in  `docusaurus.config.js` of your site.

> Then you add it in your site's `docusaurus.config.js`'s plugins option:

Reviewer: ```suggestion
Then you need to add it to the plugin options in  `docusaurus.config.js` of your site:
```

```js
module.exports = {
  // ...
  plugins: ["@edno/docusaurus2-graphql-doc-generator"],
};
```

## Configuration

You can define some or all of the plugin options directly at the plugin level in your site's `docusaurus.config.js`:

#### Plugin Options

```js
module.exports = {
  // ...
  plugins: [
    [
      "@edno/docusaurus2-graphql-doc-generator",
      {
        schema: "https://swapi.graph.cool/",
        rootPath: "./docs", // docs will be generated under './docs/swapi' (rootPath/baseURL)
        baseURL: "swapi",
        homepage: "./docs/swapi.md",
      },
    ],
};
```

> Each option is described in the section [Options](#options).

Reviewer: All the options are described in the [Options](#options) section.

> Each option is described in the section [Options](#options).

Reviewer: ```suggestion
All the options are described in the [Options](#options) section.
```
